pkg/distributor/ratestore.go

```diff
@@ -244,7 +264,7 @@ func (s *rateStore) RateFor(tenant string, streamHash uint64) int64 {
 	s.rateLock.RLock()
 	defer s.rateLock.RUnlock()
 
-	return s.rates[key(tenant, streamHash)]
+	return s.rates[key(tenant, streamHash)].rate
```

Comment:
double checking: there's no way `streamHash` won't have an entry in s.rates, right? because if it is possible, it means invoking `.rate` will panic, so a nilcheck would be necessary

Reply:
`expiringRate` is a value not a pointer so this would just return 0.

Reply:
Here is an example: https://go.dev/play/p/DLx0w-K2d9A

Reply:
ops, thats right. resolved!